google/cloud/aiplatform/helpers/container_uri_builders.py

```diff
@@ -0,0 +1,109 @@
+                framework="tensorflow",
+                framework_version="2.6",
+                accelerator="gpu"
+        )
+
+        model = aiplatform.Model.upload(
+            display_name="boston_housing_",
+            artifact_uri="gs://my-bucket/my-model/",
+            serving_container_image_uri=uri
+        )
+    ```
+
+    Args:
+        framework (str):
+            Required. The ML framework of the pre-built container. For example,
+            `"tensorflow"`, `"xgboost"`, or `"sklearn"`
+        framework_version (str):
+            Required. The version of the specified ML framework as a string.
+        region (str):
+            Optional. AI region or multi-region. Used to select the correct
+            Artifact Registry multi-region repository and reduce latency.
+            Must start with `"us"`, `"asia"` or `"europe"`.
+            Default is location set by `aiplatform.init()`.
+        accelerator (str):
+            Optional. The type of accelerator support provided by container. For
+            example: `"cpu"` or `"gpu"`
+            Default is `"cpu"`.
+
+    Returns:
+        uri (str):
+            A Vertex AI prediction container URI
+
+    Raises:
+        ValueError: If containers for provided framework are unavailable or the
+        container does not support the specified version, accelerator, or region.
+    """
+    URI_MAP = prediction._SERVING_CONTAINER_URI_MAP
+    DOCS_URI_MESSAGE = (
+        f"See {prediction._SERVING_CONTAINER_DOCUMENTATION_URL} "
+        "for complete list of supported containers"
+    )
+
+    # If region not provided, use initializer location
+    region = region or initializer.global_config.location
+    region = region.split("-", 1)[0]
+    framework = framework.lower()
+
+    if not URI_MAP.get(region):
+        raise ValueError(
+            f"Unsupported container region `{region}`, supported regions are "
+            f"{', '.join(URI_MAP.keys())}. "
+            f"{DOCS_URI_MESSAGE}"
+        )
+
+    if not URI_MAP[region].get(framework):
+        raise ValueError(
+            f"No containers found for framework `{framework}`. Supported frameworks are "
+            f"{', '.join(URI_MAP[region].keys())} {DOCS_URI_MESSAGE}"
+        )
+
+    if not URI_MAP[region][framework].get(accelerator):
+        raise ValueError(
+            f"{framework} containers do not support `{accelerator}` accelerator. Supported accelerators "
+            f"are {', '.join(URI_MAP[region][framework].keys())}. {DOCS_URI_MESSAGE}"
+        )
+
+    final_uri = URI_MAP[region][framework][accelerator].get(framework_version)
+
+    if not final_uri:
+        raise ValueError(
+            f"No serving container for `{framework}` version `{framework_version}` "
+            f"with accelerator `{accelerator}` found. Supported versions "
+            f"include {', '.join(URI_MAP[region][framework][accelerator].keys())}. {DOCS_URI_MESSAGE}"
```

Comment:
This code seems to raise error when the user calls the function with version which is already supported by backend, but not yet supported by the installed SDK.
I was asked to mark such case as not an error:
https://github.com/googleapis/python-aiplatform/pull/779#discussion_r737706240